### PR TITLE
gluster-ansible-repositories :

### PR DESCRIPTION
**Pull request by Sheersha-jain**
variable validation of gluster-ansible-repositories

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 ---
+- name: check if gluster_repos_activationkey is set
+  fail:
+    msg: "gluster_repos_activationkey is needed for enabling the repositories"
+  when: gluster_repos_activationkey is not defined
```
Comment (sabose) [inline tasks/main.yml]: shouldn't this be checked only for tags:activationkey?

File: tasks/main.yml [hunk 2]
```diff
@@ -46,6 +71,21 @@
   tags:
     - enablerepos
 
+- name: check if gluster_repos_hci_subscribe is set
+  fail:
+    msg: "Please provide HCI repositories to be attached"
+  when: gluster_repos_hci_subscribe is not defined
```
Comment (sabose) [inline tasks/main.yml]: gluster_repos_hci_subscribe - is this mandatory?

File: tasks/main.yml [hunk 3]
```diff
@@ -46,6 +71,21 @@
...
+
+- name: check if gluster_repos_nfsganesha_subscribe is set
+  fail:
+    msg: "Please provide List of nfs-ganesha repositories to attach"
+  when: gluster_repos_nfsganesha_subscribe is not defined
```
Comment (sabose) [inline tasks/main.yml]: same comment as earlier repo

File: tasks/main.yml [hunk 4]
```diff
@@ -46,6 +71,21 @@
...
+
+- name: check if gluster_repos_smb_subscribe is set
+  fail:
+    msg: "Please provide List of SMB repositories to attach"
+  when: gluster_repos_smb_subscribe is not defined
```
Comment (sabose) [inline tasks/main.yml]: same

File: tasks/main.yml [hunk 5]
```diff
@@ -37,6 +57,11 @@
   when: gluster_repos_disable_all is defined and gluster_repos_disable_all
   tags: disablerepos
 
+- name: check if gluster_repos_rhsmrepos is set
+  fail:
+    msg: "Please provide List of repositories to enable"
+  when: gluster_repos_rhsmrepos is not defined
```
Comment (sabose) [inline tasks/main.yml]: only for tags:enablerepo ?

**Review by sabose — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 1 above)
Comment (Sheersha-jain) [reply tasks/main.yml]: Do you want a tag "tags:activationkey" should be added here ?

File: tasks/main.yml (hunk 5 above)
Comment (Sheersha-jain) [reply tasks/main.yml]: Fixed

File: tasks/main.yml (hunk 2 above)
Comment (Sheersha-jain) [reply tasks/main.yml]: removed gluster_repos_hci_subscribe.

File: tasks/main.yml (hunk 3 above)
Comment (Sheersha-jain) [reply tasks/main.yml]: removed validation for this variable

File: tasks/main.yml (hunk 4 above)
Comment (Sheersha-jain) [reply tasks/main.yml]: removed validation for this variable

File: tasks/main.yml (hunk 1 above)
Comment (Sheersha-jain) [reply tasks/main.yml]: fixed. added conditional comparison between username, password and activation_key.

**Review by sac — APPROVED** [review]

**Pull request merged**